runtime/src/comm/gasnet/comm-gasnet-ex.c

```diff
@@ -714,7 +714,15 @@ int chpl_comm_addr_gettable(c_nodeid_t node, void* start, size_t len)
 
 
 int32_t chpl_comm_getMaxThreads(void) {
-  return GASNETI_MAX_THREADS-1;
+  uint64_t maxGasnetThreads = gex_System_QueryMaxThreads();
```

Comment:
@bradcray Thanks for noticing and fixing this closely related defect! 

For the record: the previous code was (mostly) correct for the default GASNet configuration where the soft and hard client thread limits are equally 256 (it was also the only option back in GASNet-1 days when this code was initially written).  Now that this PR raises the hard limit at configure time, the soft and hard limits are no longer equal so the difference matters, and this GASNet-EX query returns this process's soft limit that is in force at runtime, which is what we actually need this function to return.

One minor quibble: 
IIUC The prior code used `GASNETI_MAX_THREADS-1` because the `-1` was reserving one thread (the primordial one that runs `chpl_comm_init()`) for use by the Chapel runtime, leaving the rest available for creation by the tasking layer (the consumer of the value we are returning here). I suspect we still need to reserve that one thread to avoid an off-by-one error in our accounting of total threads that might eventually invoke a GASNet call.

TL;DR:

```suggestion
  uint64_t maxGasnetThreads = gex_System_QueryMaxThreads() - 1;
```

Reply:
This is resolved (in a slightly different manner) by 465e173